connector_search_engine/models/se_backend.py

```diff
@@ -38,9 +38,10 @@ def _get_specific_backend_selection(self):
         vals = []
         s = self.with_context(active_test=False)
         for model, descr in spec_backend_selection:
-            records = s.env[model].search([])
-            for record in records:
-                vals.append((model, record.id))
+            if model in s.env:
```

Comment:
This fix makes sense. IMO, when the upgrade is launched from the command line, the modules are imported into the python runtime and upgraded one by one during the startup process. That means that when the addon 'connector_search_engine' is upgraded, 'connector_algolia' is not yet imported into the python runtime and the values from the selection field are not yet modified by this last one. If the upgrade is launched from a fully initialized server, `connector_algolia` is already imported into the python runtime but these models are not yet registered into the odoo registry.